src/WebJobs.Extensions.DurableTask/ILoggerExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// ILogger extension methods.
+    /// </summary>
+    public static class ILoggerExtensions
+    {
+        /// <summary>
+        /// Log Information only once when the IsReplaying is set to false.
+        /// </summary>
+        /// <param name="logger">Instance of <see cref="ILogger"/></param>
+        /// <param name="contextBase">Instance of <see cref="DurableOrchestrationContextBase"/></param>
+        /// <param name="message">Information message</param>
+        public static void LogInformationOnce(this ILogger logger, DurableOrchestrationContextBase contextBase, string message)
```

Comment:
I'm not comfortable adding new public APIs if we think they might be temporary. I also share the other concerns you mentioned. Let's discuss a little bit more to see if we can find a way to change the behavior of the existing `ILogger`.

Reply:
Thank you! I will remove it.